src/Components/CreateImageWizardV2/common/TypeAheadSelect.tsx

```diff
@@ -0,0 +1,390 @@
+    }
+  };
+
+  const onInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    const enabledMenuItems = selectOptions.filter(
+      (option) => !option.isDisabled
+    );
+    const [firstMenuItem] = enabledMenuItems;
+    const focusedItem = focusedItemIndex
+      ? enabledMenuItems[focusedItemIndex]
+      : firstMenuItem;
+
+    switch (event.key) {
+      // Select the first available option
+      case 'Enter':
+        if (isOpen && focusedItem.value !== 'no results') {
+          setInputValue(String(focusedItem.children));
+          setFilterValue('');
+          if (selectedTypeIsdStringOnly) {
+            setSelected(focusedItem.value as string);
+          } else {
+            setSelected([focusedItem.value as T, String(focusedItem.children)]);
+          }
+        }
+
+        setIsOpen((prevIsOpen) => !prevIsOpen);
+        setFocusedItemIndex(null);
+        setActiveItem(null);
+
+        break;
+      case 'Tab':
+      case 'Escape':
+        setIsOpen(false);
+        setActiveItem(null);
+        break;
+      case 'ArrowUp':
+      case 'ArrowDown':
+        event.preventDefault();
+        handleMenuArrowKeys(event.key);
+        break;
+    }
+  };
+
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle
+      ref={toggleRef}
+      variant="typeahead"
+      onClick={onToggleClick}
+      isExpanded={isOpen}
+      isFullWidth
+    >
+      <TextInputGroup isPlain>
+        <TextInputGroupMain
+          value={inputValue}
+          onClick={onToggleClick}
+          onChange={onTextInputChange}
+          onKeyDown={onInputKeyDown}
+          aria-label={fieldID + 'typeahead-select-input'}
+          autoComplete="off"
+          innerRef={textInputRef}
+          placeholder={placeholderText}
+          {...(activeItem && { 'aria-activedescendant': activeItem })}
+          role="combobox"
+          isExpanded={isOpen}
+          aria-controls={fieldID + 'typeahead-select-input'}
+        />
+
+        <TextInputGroupUtilities>
+          {!!inputValue && (
+            <Button
+              variant="plain"
+              onClick={() => {
+                if (selectedTypeIsdStringOnly) {
+                  setSelected('');
+                } else {
+                  setSelected([0 as T, '']);
+                }
+                setInputValue('');
+                setFilterValue('');
+                textInputRef?.current?.focus();
+              }}
+              aria-label="Clear input value"
+            >
+              <TimesIcon aria-hidden />
+            </Button>
+          )}
+        </TextInputGroupUtilities>
+      </TextInputGroup>
+    </MenuToggle>
+  );
+
+  return (
+    <Select
+      id={fieldID}
+      isOpen={isOpen}
+      selected={selectedKey}
+      onSelect={onSelect}
+      onOpenChange={() => {
+        setIsOpen(false);
+      }}
+      toggle={toggle}
+      aria-disabled={isError}
+    >
+      <SelectList id={fieldID + '-listbox'}>
```

Comment:
The select should be disabled when the Sources service is unavailable.

![image](https://github.com/RedHatInsights/image-builder-frontend/assets/49452678/87eebf4c-454e-4ede-8d9e-3f5cca348a5b)

Reply:
And bumping this one as well :up: